doc/source/whatsnew/v0.21.0.txt

```diff
@@ -79,6 +79,62 @@ Other Enhancements
 - :func:`date_range` now accepts 'YS' in addition to 'AS' as an alias for start of year (:issue:`9313`)
 - :func:`date_range` now accepts 'Y' in addition to 'A' as an alias for end of year (:issue:`9313`)
 
+.. _whatsnew_0210.enhancements.read_csv:
+
+``read_csv`` use `python-requests` (if installed) to support basic auth and much more
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+If `python-requests` library is installed try to use it first. If not, continue using urllib
```

Comment:
>  so we don't want to introduce any inconsistencies by using a new library to handle requests for the time being

@gfyoung  Ok.. I totally agree with that specific part reasoning. And that is why earlier I had asked the question of `acceptance-criteria`. So I agree it should not use requests by default, only for the unknown/destabilization issues. But we need some way to explicitly allow use of requests in many cases and not just limit it to scenario of basic-auth.  So I propose instead of detecting only `auth` scenario, can we change it to: 
```python
if url_params is None:
    # only use existing code-base
    # never attempt to use requests even if installed... 
    # existing codebase should continue to use urllib
else: 
    # try to use requests if installed, even if it is empty. 
    # provide a way for early adopters to try out requests library even for existing scenarios

```
In this scenario, old codebase is not impacted. Only brave souls who specify the new `url_params` and if requests is installed, will use requests.
Perhaps we can set '_The acceptance criteria for using requests by default if installed - is that we get 3-6 months of user feedback on using `url_params\requests` satisfactorily_' ? Thoughts?
```python
df = pd.read_csv('http://aaa.com/joker.csv') # never uses requests. always uses `urllib`
df = pd.read_csv('http://aaa.com/joker.csv', url_params={}) # only uses requests if installed
df = pd.read_csv('http://aaa.com/joker.csv', url_params={ 'auth':('u', 'p')}) # requires requests. fails otherwise.
df = pd.read_csv('http://user:pwd@aaa.com/joker.csv', url_params={}) # only uses requests if installed
df = pd.read_csv('http://user:pwd@aaa.com/joker.csv') # Should we parse URL and look for it? Code is already written, but one can argue, lets not check in code that needs to parse url to detect auth scenario. Thoughts?
```